### PR TITLE
For #3546. Improve estimated tx size for Sapling outputs.

### PR DESCRIPTION
**Pull request by bitcartel**

### CODE REVIEW

**Comment by bitcartel** [comment]
@zkbot try

**Comment by bitcartel** [comment]
Transient failure, perhaps network related: `requested URL returned error: 404 Not Found`

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -3786,14 +3786,18 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
...
     size_t txsize = 0;
     for (int i = 0; i < zaddrRecipients.size(); i++) {
-        // TODO Check whether the recipient is a Sprout or Sapling address
-        JSDescription jsdesc;
-
-        if (mtx.fOverwintered && (mtx.nVersion >= SAPLING_TX_VERSION)) {
-            jsdesc.proof = GrothProof();
+        auto address = std::get<0>(zaddrRecipients[i]);
+        auto res = DecodePaymentAddress(address);
+        bool toSapling = boost::get<libzcash::SaplingPaymentAddress>(&res) != nullptr;
```
Comment (str4d) [inline src/wallet/rpcwallet.cpp]: Non-blocking nit: we should use a `boost::static_visitor` for this.

**Comment by bitcartel** [comment]
@zkbot r+

**Pull request merged**